cmake/FindOpenCL.cmake

```diff
@@ -42,26 +42,56 @@ if(opencl_root_hints)
 endif()
 
 function(_FIND_OPENCL_VERSION)
-  include(CheckSymbolExists)
-  include(CMakePushCheckState)
-  set(CMAKE_REQUIRED_QUIET ${OpenCL_FIND_QUIETLY})
-
-  CMAKE_PUSH_CHECK_STATE()
-  foreach(VERSION "2_2" "2_1" "2_0" "1_2" "1_1" "1_0")
-    set(CMAKE_REQUIRED_INCLUDES "${OpenCL_INCLUDE_DIR}")
-
-    if(APPLE)
-      CHECK_SYMBOL_EXISTS(
-        CL_VERSION_${VERSION}
-        "${OpenCL_INCLUDE_DIR}/Headers/cl.h"
-        OPENCL_VERSION_${VERSION})
+  foreach(VERSION "3_0" "2_2" "2_1" "2_0" "1_2" "1_1" "1_0")
```

Comment:
(random place)

There are a few things that look suspicious to me:
* We didn't implement FindOpenCL.cmake module, we've taken it from CMake (see [here](https://github.com/Kitware/CMake/blob/master/Modules/FindOpenCL.cmake)) and it still uses `CHECK_SYMBOL_EXISTS` and the whole community seems to be fine with it
* The version macros are standardized by the OpenCL specification so if the macros are missing then it likely means that there is something wrong with the OpenCL implementation
* It should be fine to check `cl.h` header as it includes `cl_version.h` header and `CHECK_SYMBOL_EXISTS` does perform a preprocessing 

Given all this, I think the issue you encounter is somewhere on your end.

Reply:
@densamoilov I have figured out that `CHECK_SYMBOL_EXISTS` is broken on 3.27.7 version, but it is working on 3.30.2

`cl.h` file I use is completely fine, it is OpenCL version for NVIDIA GPU, see file in attachments below:

[cl_files.zip](https://github.com/user-attachments/files/16529025/cl_files.zip)

Reply:
Here is my OpenCL files for NVIDIA GPU:
[cl_files.zip](https://github.com/user-attachments/files/16529151/cl_files.zip)